libflux/core/src/parser/mod.rs

```diff
@@ -1612,14 +1617,15 @@ impl Parser {
                         }
                         _ => None,
                     };
-                    match expression {
-                        Expression::Bad(_) => {
-                            items.push(ArrayItem { expression, comma });
+                    items.push(ArrayItem { expression, comma });
+                    if let Some(tok) = &self.t {
```

Comment:
You should use `self.peek()` here, as it will ensure the next token is scanned. This is guaranteed to refer to the first token that has not been consumed by the parser. The `self.t` is simply a buffer var and should not be referenced outside of the `scan`, `peek`, etc functions.